File: common/src/main/java/com/ssblur/scriptor/helpers/generators/TokenGenerator.java
package com.ssblur.scriptor.helpers.generators;

import com.google.gson.JsonObject;

import javax.annotation.Nullable;

public abstract class TokenGenerator {
  abstract public String generateToken(String key, @Nullable JsonObject parameters);
  abstract public boolean canBeDefault();

  public interface TokenGeneratorGenerator {
    TokenGenerator create(JsonObject parameters);
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/helpers/generators/TokenGenerator.kt
package com.ssblur.scriptor.helpers.generators

import com.google.gson.JsonObject

abstract class TokenGenerator {
  abstract fun generateToken(key: String, parameters: JsonObject?): String
  abstract fun canBeDefault(): Boolean

  fun interface TokenGeneratorGenerator {
    fun create(parameters: JsonObject): TokenGenerator
  }
}
